crates/nu-command/src/filesystem/cp.rs

```diff
@@ -45,311 +55,23 @@ impl Command for Cp {
     fn signature(&self) -> Signature {
         Signature::build("cp")
             .input_output_types(vec![(Type::Nothing, Type::Nothing)])
-            .required("source", SyntaxShape::GlobPattern, "the place to copy from")
-            .required("destination", SyntaxShape::Filepath, "the place to copy to")
+            .switch("recursive", "copy directories recursively", Some('r'))
+            .switch("verbose", "explicitly state what is being done", Some('v'))
             .switch(
-                "recursive",
-                "copy recursively through subdirectories",
-                Some('r'),
+                "force",
+                "if an existing destination file cannot be opened, remove it and try
+                    again (this option is ignored when the -n option is also used).
+                    currently not implemented for windows",
+                Some('f'),
             )
-            .switch(
-                "verbose",
-                "show successful copies in addition to failed copies (default:false)",
-                Some('v'),
-            )
-            .switch("update",
-                "copy only when the SOURCE file is newer than the destination file or when the destination file is missing",
-                Some('u')
-            )
-            // TODO: add back in additional features
-            // .switch("force", "suppress error when no file", Some('f'))
-            .switch("interactive", "ask user to confirm action", Some('i'))
-            .switch(
-                "no-symlink",
-                "no symbolic links are followed, only works if -r is active",
-                Some('n'),
-            )
-            .switch("progress", "enable progress bar", Some('p'))
+            .switch("interactive", "ask before overwriting files", Some('i'))
+            .switch("progress", "display a progress bar", Some('p'))
+            .switch("no-clobber", "do not overwrite an existing file", Some('n'))
+            .switch("debug", "explain how a file is copied. Implies -v", None)
+            .rest("paths", SyntaxShape::Filepath, "the place to copy to")
```

Comment:
So the help string for this is a bit tricky, because this is both "to copy from" and "to copy to". Ideally `...SRC DEST` would be possible, but I suppose it isn't? We could also make the arguments `SRC DEST` and then document `...rest` as a special case? I'm not really sure how to best encode this in the nushell parser.

Reply:
AFAIK, we have to leave it like this, because if we specify `.optional( "SRC")` and `DEST` separately, then `SRC` can only be a single source, not multiple.  Otherwise it would have to be `.optional( "SRC")`, and `.rest("TARGET")`, which could potentially include a `SRC`, so probably for now best I think to leave as it is

Reply:
Yeah, I agree. This is the best encoding at the moment.